Automattic/jetpack
# Connection: Display proper error message for suspended sites on site registration

## Proposed changes:
Display proper error message on site registration if site has been suspended instead of the generic one ("An error occurred. Please try again").

### Other information:


## Jetpack product discussion
https://github.com/Automattic/vulcan/issues/396

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Rebuild Jetpack (or My Jetpack if you're testing with it).
2. Suspend a site (see D155479-code for details).
3. Try to connect the site via Jetpack or My Jetpack. Confirm the error message says "This site has been suspended."
